### PR TITLE
file_authorization_update

### PR DESCRIPTION
**Pull request by Paul-Bob**
~~Breaking change is not a breaking change after https://github.com/avo-hq/avo/pull/1627~~

~~Deprecate general attachments authorization and~~ info about how them work together.

## Breaking change and final approach:
[another_approach.webm](https://user-images.githubusercontent.com/69730720/230039233-ee3a248c-8746-4b1b-acbb-ef87281888d6.webm)

### Second approach docs
[attachments.webm](https://user-images.githubusercontent.com/69730720/225142589-860101b4-c48c-4859-8e24-eb2390cbb268.webm)

### Third approach docs
[new_attachment_approach.webm](https://user-images.githubusercontent.com/69730720/229940477-9825f9b7-ace5-4d81-9bc7-c8242d462829.webm)

### CODE REVIEW

File: docs/2.0/upgrade.md [hunk 1]
```diff
@@ -28,39 +35,6 @@ Avo.configure do |config|
   end
 ```
 
-## Upgrade from 2.27 to 2.28
```
Comment (adrianthedev) [inline docs/2.0/upgrade.md:31]: This should still be present IMHO. Maybe someone doesn't upgrade tot he latest version, or they need the historic data.

File: docs/2.0/upgrade.md [hunk 2]
```diff
@@ -4,6 +4,13 @@ We generally push changes behind the scenes, so you don't have to update your co
 
 Follow these guides to make sure your configuration files are up to date.
 
+## Upgrade from 2.29 to 2.30
+
+We have discovered that assigning different weights to two methods that police the same thing can be a complex and challenging task. As a result `upload_attachments?`, `download_attachments?`, and `delete_attachments?` become obsolete.
+You can safely delete it from all policies.
```
Comment (adrianthedev) [inline docs/2.0/upgrade.md:10]:
```suggestion
You can safely delete it from all policies.

You may now introduce field-level authorization methods such as `upload_avatar?`, `download_avatar?`, and `delete_avatar?` if you have an `avatar` `file` field.
```

Comment (adrianthedev) [reply docs/2.0/upgrade.md:10]: We told them what to remove, we should give them a TL;DR; version of what to do next before sending them to the big docs.

File: docs/2.0/upgrade.md [hunk 3]
```diff
@@ -4,6 +4,13 @@ We generally push changes behind the scenes, so you don't have to update your co
...
+
+Go to [attachments policies methods](./../2.0/authorization.html#attachments) to apply the new methods on your policies.
```
Comment (adrianthedev) [inline docs/2.0/upgrade.md:12]:
```suggestion
More info on the [attachment authorization page](./../2.0/authorization.html#attachments).
```

File: docs/2.0/authorization.md [hunk 4]
```diff
@@ -253,28 +273,6 @@ end
 
 Now, whatever action you take for one comment, it will be available for the `edit_comments?` method in `PostPolicy`.
 
-## Files
```
Comment (adrianthedev) [inline docs/2.0/authorization.md:256]: I think we should somehow specify that the general methods have been removed in 2.31.

I would also leave the old docs specifying that we removed it in 2.31. It's important that folks don't have "dissapearing" sections.

**Review by adrianthedev — COMMENTED** [review]
I left a couple of suggestions. It's important that we keep the historical data.

File: docs/2.0/upgrade.md (hunk 1 above)
Comment (Paul-Bob) [reply docs/2.0/upgrade.md:31]: This upgrade guide is what we write for first approach, when both attachments and field level authorization was required. Before the deploy we did an commit where we loose the authorization, than only 1 of the mentioned above where needed. That loosing commit removed the breaking change and this never was truly a breaking change

File: docs/2.0/authorization.md (hunk 4 above)
Comment (Paul-Bob) [reply docs/2.0/authorization.md:256]: Totally agree, it's not the first time I do this mistake. I have a suggestion, thinking long-term I believe that filling the docs with warnings and info about changes along different versions is not scalable enough and we can easily "over-populate" the docs with this warnings. Is there any easy way to organize the docs per version? Lets say I'm using v 2.1 and you v 2.23, when I access the docs would be great to have a drop-down to select my version and see the docs for that version. Most of the content would be the same for all versions but specific content would change along the versions.

**Comment by adrianthedev** [comment]
Closing in favor of https://github.com/avo-hq/avodocs/commit/684d348d49ece8d9f1029f530b353ae72540b2a1

**Pull request closed without merging**